zcash_client_sqlite/src/prepared.rs

```diff
@@ -119,22 +165,26 @@ impl<'a, P> DataConnStmtCache<'a, P> {
                         received_by_account, address,
                         prevout_txid, prevout_idx, script,
                         value_zat, height)
-                    VALUES (
-                        :received_by_account, :address,
+                    SELECT
+                        addresses.account, :address,
                         :prevout_txid, :prevout_idx, :script,
-                        :value_zat, :height)
+                        :value_zat, :height
+                    FROM addresses
+                    WHERE addresses.cached_transparent_receiver_address = :address
```

Comment:
If `:address` does not exist in `addresses`, what happens? `utxos.received_by_account` is `NOT NULL`, so that would cause an error, but does it reach that point if the `SELECT` returns nothing due to the `WHERE` clause filtering everything out? It would be bad if this silently failed to insert.